eos-tech-support/eos-write-live-image

```diff
@@ -0,0 +1,291 @@
+        echo "$command is not installed... aborting!" >&2
+        echo "Try 'sudo apt-get install ${dependencies[$command]}'" >&2
+        exit 1
+    fi
+done
+
+if [ ! -b "$DEVICE" ]; then
+    echo "$DEVICE does not exist or is not a block device... aborting!" >&2
+    exit 1
+fi
+
+if grep -qs "$DEVICE" /proc/mounts; then
+    # Protect against overwriting the device currently in use
+    echo "$DEVICE is currently in use -- please unmount and try again" >&2
+    exit 1
+fi
+
+if [ ! "$FORCE" ]; then
+    read -p "Are you sure you want to overwrite all data on $DEVICE? [y/N] "
+    response="${REPLY,,}" # to lower
+    if [[ ! "$response" =~ ^(yes|y)$ ]]; then
+        exit 1
+    fi
+fi
+
+if [ ! -z "$FETCH_LATEST" ]; then
+    if [ -z "$OS_IMAGE" ]; then
+        OS_IMAGE=$($EOS_DOWNLOAD_IMAGE --product eos)
+    fi
+    if [ -z "$WINDOWS_TOOL" ]; then
+        WINDOWS_TOOL=$($EOS_DOWNLOAD_IMAGE --windows-tool)
+    fi
+fi
+
+if [ ! -f "$WINDOWS_TOOL" ] || [ ! -f "$OS_IMAGE" ]; then
+    echo "--os-image and --windows-tool are required if --latest is not specified" >&2
+    usage >&2
+    exit 2
+fi
+
+check_exists "$OS_IMAGE" "image"
+
+EXTRACTED_SIGNATURE="${OS_IMAGE%.gz}.asc"
+check_exists "$EXTRACTED_SIGNATURE" "uncompressed image signature"
+
+BOOT_ZIP="${OS_IMAGE%.img.gz}.boot.zip"
+check_exists "$BOOT_ZIP" "bootloader bundle"
+check_exists "$BOOT_ZIP.asc" "bootloader bundle signature"
+
+check_exists "$WINDOWS_TOOL" "Windows tool"
+
+echo
+echo "Partitioning $DEVICE..."
+
+find_by_type() {
+    echo "$1" | grep -i "type=$2" | cut -d' ' -f1
+}
+
+if [ "$MBR" ]; then
+    sfdisk --label dos "$DEVICE" <<MBR_PARTITIONS
+1 : type=7
+MBR_PARTITIONS
+    udevadm settle
+    partprobe "$DEVICE"
+    PARTMAP=$(sfdisk --dump "$DEVICE")
+    DEVICE_IMAGES=$(find_by_type "$PARTMAP" "7")
+else
+    # https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_type_GUIDs
+    PARTITION_SYSTEM_GUID="c12a7328-f81f-11d2-ba4b-00a0c93ec93b"
+    PARTITION_BIOS_BOOT_GUID="21686148-6449-6E6F-744E-656564454649"
+    PARTITION_BASIC_DATA_GUID="ebd0a0a2-b9e5-4433-87c0-68b6b72699c7"
+
+    # We want the data partition, "eoslive", to occupy all the space on the disk
+    # after the UEFI and BIOS boot partitions. But, we also want it to be numbered
+    # first: apparently Windows will only mount the partition numbered first,
+    # regardless of where it is on the disk.
+    #
+    # It is important that the offset of the BIOS boot partition matches that
+    # used in the Endless OS image builder, since it is embedded in the GRUB
+    # image.
+    sfdisk --label gpt "$DEVICE" <<EFI_PARTITIONS
+2 : start=2048, size=62MiB, type=$PARTITION_SYSTEM_GUID
+3 : size=1MiB, type=$PARTITION_BIOS_BOOT_GUID
+1 : name=eoslive, type=$PARTITION_BASIC_DATA_GUID
+EFI_PARTITIONS
+    udevadm settle
+    partprobe "$DEVICE"
+    PARTMAP=$(sfdisk --dump "$DEVICE")
+    DEVICE_IMAGES=$(find_by_type "$PARTMAP" "$PARTITION_BASIC_DATA_GUID")
```

Comment:
Yeah, seems good. I wish there was a more straightforward way to do this. Here's another way I found with `lsblk`:
```
$ DEVICE=/dev/sda
$ GUID=c12a7328-f81f-11d2-ba4b-00a0c93ec93b
$ part=$(lsblk -o PARTTYPE,NAME -nl "$DEVICE" | awk "{if (\$1 == \"$GUID\") print \$2}")
$ echo $part
sda1
```